crates/interfaces/src/provider.rs

```diff
@@ -10,6 +10,11 @@ pub enum Error {
     BlockHash { block_hash: BlockHash },
     #[error("Block body not exists #{block_number} ({block_hash:?})")]
     BlockBody { block_number: BlockNumber, block_hash: BlockHash },
-    #[error("Block transition does not exist for block #{block_number} ({block_hash:?})")]
-    BlockTransition { block_number: BlockNumber, block_hash: BlockHash },
+    #[error("Block transition id does not exist for block #{block_number}")]
+    BlockTransition { block_number: BlockNumber },
+
+    #[error("Block number {block_number} does not exist in canonical chain")]
+    BlockCanonical { block_number: BlockNumber },
```

Comment:
this variant is used as an error in the same place as `Error::BlockNumber`. can we dedup?

Reply:
It is different error. This means hash is correct but for any reason it can be found in canonical chain. I included block hash inside BlockCanoniacl error for more info.